matter/wifi/rs911x/rsi_if.c

```diff
@@ -545,23 +549,18 @@ void wfx_rsi_task(void *arg)
 #endif /* SL_WFX_CONFIG_SOFTAP */
   }
 }
-void wfx_dhcp_got_ipv4(uint32_t ip)
-{
+void wfx_dhcp_got_ipv4(uint32_t ip) {
   /*
    * Acquire the new IP address
    */
-  wfx_rsi.ip4_addr[0] = (ip)&0xff;
-  wfx_rsi.ip4_addr[1] = (ip >> 8) & 0xff;
-  wfx_rsi.ip4_addr[2] = (ip >> 16) & 0xff;
-  wfx_rsi.ip4_addr[3] = (ip >> 24) & 0xff;
-  WFX_RSI_LOG("%s: DHCP OK: IP=%d.%d.%d.%d",
-              __func__,
-              wfx_rsi.ip4_addr[0],
-              wfx_rsi.ip4_addr[1],
-              wfx_rsi.ip4_addr[2],
-              wfx_rsi.ip4_addr[3]);
+  wfx_rsi.ip4_addr[0] = (ip)& HEX_VALUE_FF;
+  wfx_rsi.ip4_addr[1] = (ip >> SHIFT_8)  & HEX_VALUE_FF;
+  wfx_rsi.ip4_addr[2] = (ip >> SHIFT_16) & HEX_VALUE_FF;
+  wfx_rsi.ip4_addr[3] = (ip >> SHIFT_24) & HEX_VALUE_FF;
+  WFX_RSI_LOG("%s: DHCP OK: IP=%d.%d.%d.%d", __func__, wfx_rsi.ip4_addr[0],
+              wfx_rsi.ip4_addr[1], wfx_rsi.ip4_addr[2], wfx_rsi.ip4_addr[3]);
```

Comment:
same comment as value set.

Reply:
kept the original implementation for now, don't want to change much now.